src/coreclr/vm/precode.cpp

```diff
@@ -182,6 +182,55 @@ PCODE Precode::TryToSkipFixupPrecode(PCODE addr)
     return 0;
 }
 
+#ifdef FEATURE_INTERPRETER
+static TADDR GetMethodDescFromPrecodeData(TADDR methodDesc)
+{
+    LIMITED_METHOD_CONTRACT;
+    if ((methodDesc & InterpretedCodeAddressFlag) == 0)
+    {
+        return methodDesc;
+    }
+
+    PTR_CodeHeader pCodeHeader = PTR_CodeHeader((methodDesc & ~(TADDR)1) - sizeof(CodeHeader));
```

Comment:
```suggestion
    PTR_CodeHeader pCodeHeader = PTR_CodeHeader((methodDesc & ~(TADDR)InterpretedCodeAddressFlag) - sizeof(CodeHeader));
```

Did you mean to mask off `InterpretedCodeAddressFlag` (2) instead?

Reply:
Oops, right, thank you for spotting it.

Reply:
In fact, I should also use the EEJitManager::GetCodeHeaderFromStartAddress instead of doing the pointer math here. I've fallen into the trap of accepting what copilot has provided me with here.

Reply:
Fixed in an additional commit